jax/lax/lax_control_flow.py

```diff
@@ -287,6 +288,13 @@ def cond(pred, true_operand, true_fun, false_operand, false_fun):
       else:
         return false_fun(false_operand)
   """
+  pred_dtype = array(pred).dtype
+  if pred_dtype.kind != 'b':
+    if pred_dtype.kind in 'iuf':
```

Comment:
allow int, unsigned int, float
(maybe too much happy to narrow down)

Reply:
I think "anything that's okay in `np.where`" is probably a good guideline here. `np.where` works with all numeric types, so this looks good to me.